src/map.cpp

```diff
@@ -1983,17 +1911,11 @@ void ServerMap::loadBlock(std::string *blob, v3s16 p3d, MapSector *sector, bool
 			scanner.scan(block, &m_transforming_liquid);
 		}
 
-		/*
-			Save blocks loaded in old format in new format
-		*/
-
-		//if(version < SER_FMT_VER_HIGHEST_READ || save_after_load)
-		// Only save if asked to; no need to update version
-		if(save_after_load)
-			saveBlock(block);
-
-		// We just loaded it from, so it's up-to-date.
-		block->resetModified();
+		// Upgrade ancient mapblock versions on the fly
+		if (version < SER_FMT_VER_LOWEST_WRITE)
+			block->raiseModified(MOD_STATE_WRITE_AT_UNLOAD, MOD_REASON_SER_FMT_UPGRADE);
+		else
+			block->resetModified();
```

Comment:
Sneaky change. This updates mapblocks automatically if they're older than what `MapBlock::serialize` supports. It should spread the load and incrementally update old maps.

Is it acceptable or shall I revert this part?